File: src/main/java/model/Product.java
package model;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "Products")
public class Product {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "ProductID", nullable = false)
    private UUID id;

    @Nationalized
    @Column(name = "ProductName", nullable = false)
    private String productName;

    @Nationalized
    @Column(name = "SKU", nullable = false, length = 100)
    private String sku;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "CategoryID", nullable = false)
    private Category categoryID;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "BrandID", nullable = false)
    private Brand brandID;

    @Nationalized
    @Lob
    @Column(name = "Description")
    private String description;

    @Nationalized
    @Column(name = "ShortDescription", length = 500)
    private String shortDescription;

    @Column(name = "Price", nullable = false, precision = 18, scale = 2)
    private BigDecimal price;

    @Column(name = "ComparePrice", precision = 18, scale = 2)
    private BigDecimal comparePrice;

    @Column(name = "CostPrice", precision = 18, scale = 2)
    private BigDecimal costPrice;

    @Column(name = "Weight", precision = 10, scale = 2)
    private BigDecimal weight;

    @Nationalized
    @Column(name = "Dimensions", length = 100)
    private String dimensions;

    @ColumnDefault("0")
    @Column(name = "StockQuantity", nullable = false)
    private Integer stockQuantity;

    @ColumnDefault("5")
    @Column(name = "MinStockLevel")
    private Integer minStockLevel;

    @ColumnDefault("1000")
    @Column(name = "MaxStockLevel")
    private Integer maxStockLevel;

    @ColumnDefault("1")
    @Column(name = "IsActive")
    private Boolean isActive;

    @ColumnDefault("0")
    @Column(name = "IsFeatured")
    private Boolean isFeatured;

    @ColumnDefault("0")
    @Column(name = "ViewCount")
    private Integer viewCount;

    @ColumnDefault("0")
    @Column(name = "SalesCount")
    private Integer salesCount;

    @ColumnDefault("0")
    @Column(name = "AverageRating", precision = 3, scale = 2)
    private BigDecimal averageRating;

    @ColumnDefault("0")
    @Column(name = "ReviewCount")
    private Integer reviewCount;

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    private Instant createdDate;

    @ColumnDefault("getdate()")
    @Column(name = "ModifiedDate")
    private Instant modifiedDate;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getSku() {
        return sku;
    }

    public void setSku(String sku) {
        this.sku = sku;
    }

    public Category getCategoryID() {
        return categoryID;
    }

    public void setCategoryID(Category categoryID) {
        this.categoryID = categoryID;
    }

    public Brand getBrandID() {
        return brandID;
    }

    public void setBrandID(Brand brandID) {
        this.brandID = brandID;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getShortDescription() {
        return shortDescription;
    }

    public void setShortDescription(String shortDescription) {
        this.shortDescription = shortDescription;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public BigDecimal getComparePrice() {
        return comparePrice;
    }

    public void setComparePrice(BigDecimal comparePrice) {
        this.comparePrice = comparePrice;
    }

    public BigDecimal getCostPrice() {
        return costPrice;
    }

    public void setCostPrice(BigDecimal costPrice) {
        this.costPrice = costPrice;
    }

    public BigDecimal getWeight() {
        return weight;
    }

    public void setWeight(BigDecimal weight) {
        this.weight = weight;
    }

    public String getDimensions() {
        return dimensions;
    }

    public void setDimensions(String dimensions) {
        this.dimensions = dimensions;
    }

    public Integer getStockQuantity() {
        return stockQuantity;
    }

    public void setStockQuantity(Integer stockQuantity) {
        this.stockQuantity = stockQuantity;
    }

    public Integer getMinStockLevel() {
        return minStockLevel;
    }

    public void setMinStockLevel(Integer minStockLevel) {
        this.minStockLevel = minStockLevel;
    }

    public Integer getMaxStockLevel() {
        return maxStockLevel;
    }

    public void setMaxStockLevel(Integer maxStockLevel) {
        this.maxStockLevel = maxStockLevel;
    }

    public Boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(Boolean isActive) {
        this.isActive = isActive;
    }

    public Boolean getIsFeatured() {
        return isFeatured;
    }

    public void setIsFeatured(Boolean isFeatured) {
        this.isFeatured = isFeatured;
    }

    public Integer getViewCount() {
        return viewCount;
    }

    public void setViewCount(Integer viewCount) {
        this.viewCount = viewCount;
    }

    public Integer getSalesCount() {
        return salesCount;
    }

    public void setSalesCount(Integer salesCount) {
        this.salesCount = salesCount;
    }

    public BigDecimal getAverageRating() {
        return averageRating;
    }

    public void setAverageRating(BigDecimal averageRating) {
        this.averageRating = averageRating;
    }

    public Integer getReviewCount() {
        return reviewCount;
    }

    public void setReviewCount(Integer reviewCount) {
        this.reviewCount = reviewCount;
    }

    public Instant getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Instant createdDate) {
        this.createdDate = createdDate;
    }

    public Instant getModifiedDate() {
        return modifiedDate;
    }

    public void setModifiedDate(Instant modifiedDate) {
        this.modifiedDate = modifiedDate;
    }

}

File: src/main/java/model/Product.kt
package model

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Nationalized
import java.math.BigDecimal
import java.time.Instant
import java.util.*

@Entity
@Table(name = "Products")
open class Product {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "ProductID", nullable = false)
    open var id: UUID? = null

    @Nationalized
    @Column(name = "ProductName", nullable = false)
    open var productName: String? = null

    @Nationalized
    @Column(name = "SKU", nullable = false, length = 100)
    open var sku: String? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "CategoryID", nullable = false)
    open var categoryID: Category? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "BrandID", nullable = false)
    open var brandID: Brand? = null

    @Nationalized
    @Lob
    @Column(name = "Description")
    open var description: String? = null

    @Nationalized
    @Column(name = "ShortDescription", length = 500)
    open var shortDescription: String? = null

    @Column(name = "Price", nullable = false, precision = 18, scale = 2)
    open var price: BigDecimal? = null

    @Column(name = "ComparePrice", precision = 18, scale = 2)
    open var comparePrice: BigDecimal? = null

    @Column(name = "CostPrice", precision = 18, scale = 2)
    open var costPrice: BigDecimal? = null

    @Column(name = "Weight", precision = 10, scale = 2)
    open var weight: BigDecimal? = null

    @Nationalized
    @Column(name = "Dimensions", length = 100)
    open var dimensions: String? = null

    @ColumnDefault("0")
    @Column(name = "StockQuantity", nullable = false)
    open var stockQuantity: Int? = null

    @ColumnDefault("5")
    @Column(name = "MinStockLevel")
    open var minStockLevel: Int? = null

    @ColumnDefault("1000")
    @Column(name = "MaxStockLevel")
    open var maxStockLevel: Int? = null

    @ColumnDefault("1")
    @Column(name = "IsActive")
    open var isActive: Boolean? = null

    @ColumnDefault("0")
    @Column(name = "IsFeatured")
    open var isFeatured: Boolean? = null

    @ColumnDefault("0")
    @Column(name = "ViewCount")
    open var viewCount: Int? = null

    @ColumnDefault("0")
    @Column(name = "SalesCount")
    open var salesCount: Int? = null

    @ColumnDefault("0")
    @Column(name = "AverageRating", precision = 3, scale = 2)
    open var averageRating: BigDecimal? = null

    @ColumnDefault("0")
    @Column(name = "ReviewCount")
    open var reviewCount: Int? = null

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    open var createdDate: Instant? = null

    @ColumnDefault("getdate()")
    @Column(name = "ModifiedDate")
    open var modifiedDate: Instant? = null
}